src/scvmm/azext_scvmm/custom.py

```diff
@@ -88,26 +88,43 @@ def connect_vmmserver(
     fqdn=None,
     username=None,
     password=None,
-    port=DEFAULT_VMMSERVER_PORT,
+    port=None,
     tags=None,
     no_wait=False,
 ):
 
-    creds_ok = all(inp is not None for inp in [fqdn, username, password])
+    creds_ok = all(inp is not None for inp in [fqdn, port, username, password])
     while not creds_ok:
         creds = {
             'fqdn': fqdn,
+            'port': port,
             'username': username,
             'password': password,
         }
-        if fqdn is None:
+        while not creds['fqdn']:
             print('Please provide vmmserver FQDN or IP address: ', end='')
             creds['fqdn'] = input()
-        if username is None:
+            if not creds['fqdn']:
+                print('Parameter is required, please try again')
+        while not creds['port']:
+            print('Please provide vmmserver port (Default: 8100): ', end='')
+            try:
+                creds['port'] = input()
+                if not creds['port']:
+                    creds['port'] = DEFAULT_VMMSERVER_PORT
+                creds['port'] = int(creds['port'])
+            except ValueError:
+                print('Port must be a number, please try again')
+                creds['port'] = None
```

Comment:
For the automation scenario, if the CLI script does not pass in the `creds['port']` parameter, the process of CLI script will get stuck. Is this in line with expectations?

Reply:
Yes, this is in line with the expectations. We will add modify the command by adding the new param to our automation pipelines.